### PR TITLE
refactor: fix all clint warnings from buffer

### PR DESCRIPTION
**Pull request by dundargoc**
Sanity check to see if people still want to go with the "slow and gradual" strategy.

### CODE REVIEW

File: src/nvim/buffer.c [hunk 1]
```diff
@@ -5155,15 +5076,13 @@ void ex_buffer_all(exarg_T *eap)
...
- *
- * "flags" can be:
- * OPT_WINONLY      only set options local to window
- * OPT_NOWIN        don't set options local to window
- *
- * Returns immediately if the "ml" option isn't set.
- */
+/// do_modelines() - process mode lines for the current file
+///
+/// "flags" can be:
```
Comment (bfredl) [inline src/nvim/buffer.c:5081]: `@param flags`

**Review by zeertzjq — COMMENTED** [review]
In some places there are two spaces after `@return`, while in some other places there is only one.

File: src/nvim/buffer.c [hunk 2]
```diff
@@ -1410,15 +1400,13 @@ int do_buffer(int action, int start, int dir, int count, int forceit)
...
- * Set current buffer to "buf".  Executes autocommands and closes current
- * buffer.  "action" tells how to close the current buffer:
- * DOBUF_GOTO       free or hide it
- * DOBUF_SPLIT      nothing
- * DOBUF_UNLOAD     unload it
- * DOBUF_DEL        delete it
- * DOBUF_WIPE       wipe it out
- */
+/// Set current buffer to "buf".  Executes autocommands and closes current
+/// buffer.  "action" tells how to close the current buffer:
```
Comment (bfredl) [inline src/nvim/buffer.c:1404]: `@param action` on a new line

File: src/nvim/buffer.c [hunk 3]
```diff
@@ -2251,11 +2221,9 @@ static int buf_time_compare(const void *s1, const void *s2)
...
 }
 
-/*
- * Find all buffer names that match.
- * For command line expansion of ":buf" and ":sbuf".
- * Return OK if matches found, FAIL otherwise.
- */
+/// Find all buffer names that match.
+/// For command line expansion of ":buf" and ":sbuf".
+/// Return OK if matches found, FAIL otherwise.
```
Comment (bfredl) [inline src/nvim/buffer.c:2226]: `@return`

File: src/nvim/buffer.c [hunk 4]
```diff
@@ -2059,10 +2037,8 @@ void buflist_getfpos(void)
   }
 }
 
-/*
- * Find file in buffer list by name (it has to be for the current window).
- * Returns NULL if not found.
- */
+/// Find file in buffer list by name (it has to be for the current window).
+/// Returns NULL if not found.
```
Comment (bfredl) [inline src/nvim/buffer.c:2043]: `@return buffer or NULL if not found` (and similar for functions below)

Comment (dundargoc) [reply src/nvim/buffer.c]: I could've sworn I'd fixed all doxygen comments yesterday O_O . Thanks.

**Review by bfredl — APPROVED** [review]
Squash?

**Pull request merged**